packages/services/src/state_committer.rs

```diff
@@ -420,4 +475,56 @@ mod tests {
         // Mocks validate that the fragments have been sent despite insufficient accumulation
         Ok(())
     }
+
+    #[tokio::test]
+    async fn resubmits_fragments_when_gas_bump_timeout_exceeded() -> Result<()> {
+        // given
+        // init test clock with actual time to be able to compare with db timestamps
+        let clock = TestClock::new(SystemClock {}.now());
+        let setup = test_utils::Setup::init().await;
+
+        let fragments = setup.insert_fragments(0, 5).await;
+
+        let tx_hash_1 = [6; 32];
+        let tx_hash_2 = [7; 32];
+        let l1_mock_submit = test_utils::mocks::l1::expects_state_submissions([
+            (
+                Some(NonEmpty::from_vec(fragments.clone()).unwrap()),
+                tx_hash_1,
+            ),
+            (
+                Some(NonEmpty::from_vec(fragments.clone()).unwrap()),
+                tx_hash_2,
+            ),
+        ]);
+
+        let fuel_mock = test_utils::mocks::fuel::latest_height_is(0);
+        let mut state_committer = StateCommitter::new(
+            l1_mock_submit,
+            fuel_mock,
+            setup.db(),
+            Config {
+                lookback_window: 1000,
+                fragment_accumulation_timeout: Duration::from_secs(60),
+                fragments_to_accumulate: 5.try_into().unwrap(),
+                gas_bump_timeout: Duration::from_secs(60),
+                ..Default::default()
+            },
+            clock.clone(),
+        );
+
+        // Submit the initial fragments
+        state_committer.run().await?;
+
+        // Advance time beyond the gas bump timeout, we need to advance a bit more
```

Comment:
Having the db record the time creates difficulties for testing. We can leave it for now but it would be better if whomever submits the transaction should also give the time it is submitted at to the db insert query.